opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java

```diff
@@ -488,61 +465,71 @@ private String getFirstRevision(String fullpath) {
+            if (origName != null) {
+                annotation = getAnnotation(revision, origName);
             }
-            cmd.add("--");
-            cmd.add(findOriginalName(file.getCanonicalPath(), revision));
-            executor = new Executor(cmd, new File(getDirectoryName()),
-                    RuntimeEnvironment.getInstance().getInteractiveCommandTimeout());
-            parser = new GitAnnotationParser(file.getName());
-            status = executor.exec(true, parser);
         }
 
-        if (status != 0) {
+        return annotation;
+    }
+
+    private String getFirstRevision(String filePath) {
+        String revision = null;
+        try (org.eclipse.jgit.lib.Repository repository = getJGitRepository(getDirectoryName())) {
+            Iterable<RevCommit> commits = new Git(repository).log().
+                    addPath(getGitFilePath(filePath)).
+                    setMaxCount(1).
+                    call();
+            RevCommit commit = commits.iterator().next();
+            if (commit != null) {
+                revision = commit.getId().getName();
+            } else {
+                LOGGER.log(Level.WARNING, String.format("cannot get first revision of '%s' in repository '%s'",
+                        filePath, getDirectoryName()));
+            }
+        } catch (IOException | GitAPIException e) {
             LOGGER.log(Level.WARNING,
-                    "Failed to get annotations for: \"{0}\" Exit code: {1}",
-                    new Object[]{file.getAbsolutePath(), String.valueOf(status)});
-            return null;
+                    String.format("cannot get first revision of '%s' in repository '%s'",
+                            filePath, getDirectoryName()), e);
         }
+        return revision;
+    }
+
+    @NotNull
+    private Annotation getAnnotation(String revision, String filePath) throws IOException {
+        Annotation annotation = new Annotation(filePath);
 
-        return parser.getAnnotation();
+        try (org.eclipse.jgit.lib.Repository repository = getJGitRepository(getDirectoryName())) {
+            BlameCommand blameCommand = new Git(repository).blame().setFilePath(getGitFilePath(filePath));
+            ObjectId commitId = repository.resolve(revision);
+            blameCommand.setStartCommit(commitId);
+            blameCommand.setFollowFileRenames(isHandleRenamedFiles());
+            final BlameResult result = blameCommand.setTextComparator(RawTextComparator.WS_IGNORE_ALL).call();
+            if (result != null) {
+                final RawText rawText = result.getResultContents();
+                for (int i = 0; i < rawText.size(); i++) {
+                    final PersonIdent sourceAuthor = result.getSourceAuthor(i);
+                    final RevCommit sourceCommit = result.getSourceCommit(i);
+                    annotation.addLine(sourceCommit.getId().abbreviate(8).name(), sourceAuthor.getName(), true);
```

Comment:
Maybe create a constant for `8`?

Reply:
sure. I thought there is one however could not find it.